### PR TITLE
feat(app): Add private routes

### PR DESCRIPTION
**Pull request by sldk-yuri**
One of the point of road map is **Private routes**. This PR add guards that implement routes for auth and not authorized users. It's pretty straightforward right now, could be a good start point or ready made feature for now.

@noveogroup-amorgunov could you take a time and review this, please?

Thank you.

### CODE REVIEW

File: src/app/appRouter.tsx [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 import { LoginPage } from '@/pages/login'
 import { MainPage } from '@/pages/main'
 import { ProductPage } from '@/pages/product'
 import { WishlistPage } from '@/pages/wishlist'
+import { useAppSelector } from '@/shared/model'
 import { appStore } from './appStore'
 import { baseLayout } from './layouts/baseLayout'
 import { layoutWithSidebar } from './layouts/layoutWithSidebar'
 
+const GuestGuard = ({ children }: { children: ReactNode }) => {
```
Comment (noveogroup-amorgunov) [inline src/app/appRouter.tsx:16]: React components are written with function declaration in the project. Let's rewrite this:

```suggestion
type GuardProps = {...}

function GuestGuard({ children }: GuardProps) {
```

File: src/app/appRouter.tsx [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+  const isAuthorized = useAppSelector(selectIsAuthorized)
+
+  if (!isAuthorized) return <Navigate to="/" />
```
Comment (noveogroup-amorgunov) [inline src/app/appRouter.tsx:19]: I suggest redirect to "/login" page in this case

File: src/app/appRouter.tsx [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+
+  return <> {children} </>
+}
+
+const AuthGuard = ({ children }: { children: ReactNode }) => {
+  const isAuthorized = useAppSelector(selectIsAuthorized)
+
+  if (isAuthorized) return <Navigate to="/" />
+
+  return <> {children} </>
```
Comment (noveogroup-amorgunov) [inline src/app/appRouter.tsx:29]: Could we not use the fragment?

```suggestion
  return children
```

**Review by noveogroup-amorgunov — COMMENTED** [review]
@sldk-yuri hey! Thank you for contribution. I wrote few comments, let's fix it,  and I will merge this feat

File: src/app/appRouter.tsx [hunk 4]
```diff
@@ -1,15 +1,42 @@
...
+import { useAppSelector } from '@/shared/model'
 import { appStore } from './appStore'
 import { baseLayout } from './layouts/baseLayout'
 import { layoutWithSidebar } from './layouts/layoutWithSidebar'
 
+type GuestGuardProps = {
+  children: ReactNode
+}
+
+const GuestGuard = ({ children }: GuestGuardProps) => {
```
Comment (noveogroup-amorgunov) [inline src/app/appRouter.tsx:20]: Let's rewrire to function declarations from function expression, like

```suggestion
function GuestGuard({ children }: GuestGuardProps {
```

**Review by noveogroup-amorgunov — COMMENTED** [review]
Thx you for fixed. I left one comment, could you please fix it?. Also, let's mark checkbox (private routes) in README.md

**Comment by sldk-yuri** [comment]
@noveogroup-amorgunov should looks well for now 👍

**Review by noveogroup-amorgunov — APPROVED** [review]
🔥

**Pull request merged**